grails-app/services/au/org/ala/phyloviz/AlaService.groovy

```diff
@@ -24,16 +25,21 @@ class AlaService {
      */
 
     /**
-     * bulk LSID look up. The webservice returns null for those species it could not find an LSID.
-     * @param names
-     * @return
+     * Retrieves a list of LSIDs for the provided list of names. The resulting list will be of the same length as the
+     * provided list of names, and each cell will contain the LSID for the corresponding cell in the names list, or null
+     * if there is no matching LSID.
+     *
+     * @param names List of names to search for
+     * @return a list of LSIDs corresponding to the items in the provided names list, or null if there was no match.
      */
-    def getLsid(names) {
-        def url = grailsApplication.config.lsidBulkLookup;
-        // POST request body format
-        // '{"names":["Macropus rufus","Macropus greyi"]}'
-        names = (['names': names] as JSON).toString(true);
-        return webService.doPost(url, '', '', names)
+    List getLsid(List names) {
+        names?.collect {
+            try {
+                nameService.getLSID(it)
```

Comment:
nameService.getLSID() will return null if there is no match.

alaService.getLsid() should always return a list with the same number of elements as the input list, with each cell containing either the matched LSID or null, and the order should be consistent. I will write some tests to make sure.